### PR TITLE
Clair3 version topics migration

### LINKED ISSUE #10643: Migration to topics channels: clair3

### PR DESCRIPTION
**Pull request by EfraMP**
## PR checklist

Closes #10643, related with all modules topics migration #10832

+ There was a major clair3 update from v1.5 to v2.0, migrating from TensorFlow models to PyTorch
+ Updated said models at the [test repo](https://github.com/nf-core/test-datasets/pull/1917) 
+ Updated the used containers and new parameters usage
+ ⚠️ Note that he `packaged_model` test fails on **conda** because in clair3 v2.0.0 the DL models are no longer distributed with the bioconda package. Docker and Singularity tests pass successfully, with the models bundled in the Docker/Singularity container at `/opt/models/`

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [x] Broadcast software version numbers to `topic: versions` - [See version_topics](https://nf-co.re/blog/2025/version_topics)
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

**Comment by jfy133** [comment]
> ⚠️ Note that he packaged_model test fails on conda because in clair3 v2.0.0 the DL models are no longer distributed with the bioconda package. Docker and Singularity tests pass successfully, with the models bundled in the Docker/Singularity container at /opt/models/

Would there be a way to have an optional input channel for the models, so a user could supply them 'manually', in case they are running with conda?

**Comment by EfraMP** [comment]
Yes! It's passed as `user_model` at line 11 in the `main.nf`

**Comment by jfy133** [comment]
Ok great! Then I think you can add to one of the `.github/workflow` files the name of the module to skip the Conda tests so all tests pass on github

**Comment by EfraMP** [comment]
Done!

File: modules/nf-core/clair3/tests/main.nf.test.snap [hunk 1]
```diff
@@ -49,15 +49,21 @@
...
-            "nf-test": "0.9.2",
-            "nextflow": "25.04.6"
-        },
-        "timestamp": "2025-09-10T12:49:24.796196"
+            "nf-test": "0.9.4",
+            "nextflow": "25.10.4"
+        }
     },
     "sarscov2 - bam - packaged_model": {
         "content": [
```
Comment (kubranarci) [inline modules/nf-core/clair3/tests/main.nf.test.snap:69]: is this expected to me empty?

Comment (EfraMP) [reply modules/nf-core/clair3/tests/main.nf.test.snap:69]: Do you mean the timestamp? It moved up the meta

**Review by kubranarci — APPROVED** [review]

**Comment by SPPearce** [comment]
Hmm, this change also swapped the container to use docker.io instead of biocontainers/seqera containers.
Was there a reason for that @EfraMP ?

**Comment by EfraMP** [comment]
Oh, it's the container from the original dev. Is it a big issue???

If it's a big deal I can change the module, and bump the version in the meantime as Clair3's latest release was done a couple of hours ago.

**Comment by SPPearce** [comment]
We like to mirror the containers: https://nf-co.re/docs/specifications/pipelines/recommendations/custom_containers
